libraries/botbuilder-dialogs-adaptive/src/recognizers/adaptiveRecognizer.ts

```diff
@@ -0,0 +1,69 @@
+/**
+ * @module botbuilder-dialogs-adaptive
+ */
+/**
+ * Copyright (c) Microsoft Corporation. All rights reserved.
+ * Licensed under the MIT License.
+ */
+import { BoolExpression } from "adaptive-expressions";
+import { getTopScoringIntent, RecognizerResult } from "botbuilder-core";
+import { DialogContext, Recognizer } from "botbuilder-dialogs";
+
+// TODO
+// (differs from C# in that it uses <name-of-recognizer>Configuration interfaces)
+// AdaptiveRecognizerConfiguration interface
+// with optional BoolExpression logPersonalInformation
+
+export abstract class AdaptiveRecognizer extends Recognizer {
+    /**
+     * (Optional) Flag that designates whether personally identifiable information (PII) should log to telemetry.
+     */
+    public logPersonalInformation = new BoolExpression('=settings.telemetry.logPersonalInformation');
+
+    /**
+     * Uses the RecognizerResult to create a list of propeties to be included when tracking the result in telemetry.
+     * @param recognizerResult Recognizer Result.
+     * @param telemetryProperties A list of properties to append or override the properties created using the RecognizerResult.
+     * @param dialogContext Dialog Context.
+     * @returns A dictionary that can be included when calling the TrackEvent method on the TelemetryClient.
+     */
+    protected fillRecognizerResultTelemetryProperties(
+        recognizerResult: RecognizerResult,
+        telemetryProperties: { [key: string]: string },
+        dialogContext?: DialogContext
+    ): { [key: string]: string } {
+        if (dialogContext == null)
+        {
+            throw new Error('DialogContext needed for state in AdaptiveRecognizer.fillRecognizerResultTelemetryProperties method.');
+        }
+        const { intent, score } = getTopScoringIntent(recognizerResult);
+
+        const properties: { [key: string]: string } = {
+            TopIntent: Object.entries(recognizerResult.intents).length > 0 ? intent : undefined,
```

Comment:
You're computing this `Object.entries(...)` several times - in that case it should be computed once and stored in a `const`.

Reply:
Hmm indeed. Updated w/`const`  here and in `Recognizer` as well now ✔️